auto_nag/scripts/reminder.py

```diff
@@ -0,0 +1,79 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import re
+
+import dateutil.parser
+
+from auto_nag.bzcleaner import BzCleaner
+
+DATE_MATCH = re.compile(r"\[reminder-?[a-zA-Z0-9]* ([0-9]{4}-[0-9]{1,2}-[0-9]{1,2})\]")
+
+
+class Reminder(BzCleaner):
+    def __init__(self):
+        super(Reminder, self).__init__()
+        self.changes_per_bug = {}
+
+    def description(self):
+        return "Allow users to set reminders on bugs"
+
+    def get_bz_params(self, date):
+        params = {
+            "include_fields": ["assigned_to", "whiteboard", "triage_owner"],
+            "f1": "whiteboard",
+            "o1": "substring",
+            "v1": "[reminder",
+        }
+
+        return params
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        new_whiteboard = whiteboard = bug["whiteboard"]
+        matches = DATE_MATCH.findall(whiteboard)
+        for m in matches:
+            target = dateutil.parser.parse(m)
+            today = datetime.datetime.today()
+
+            if target.date() < today.date():
+                replace_string = f"[reminder {m}]"
+                new_whiteboard = new_whiteboard.replace(replace_string, "")
+
+        if new_whiteboard != whiteboard:
+            assignee = bug["assigned_to"]
+            triage_owner = bug["triage_owner"]
+
+            self.changes_per_bug[bugid] = {
+                "comment": {"body": self.comment_tpl_for_bugid(bugid)},
+                "whiteboard": new_whiteboard,
+                "flags": [
+                    {
+                        "name": "needinfo",
+                        "requestee": assignee,
+                        "status": "?",
+                        "new": "true",
+                    },
+                    {
+                        "name": "needinfo",
+                        "requestee": triage_owner,
+                        "status": "?",
+                        "new": "true",
+                    },
+                ],
```

Comment:
I've null-coalesced to asignee || triage_owner; but I don't think I can use `get_mail_to_auto_ni` because I don't want to set an ni for every bug; only the ones where the date is applicable?    I've updated the code, but maybe it can be simplified further using `get_mail_to_auto_ni` and a template but I'm not sure.

Reply:
When `handle_bug()` returns `None`, the bug will not be included when using `get_mail_to_auto_ni()`.

https://github.com/mozilla/relman-auto-nag/blob/032d00b21faebc2cc71ab8d57ebc6194b5185b24/auto_nag/scripts/reminder.py#L50-L51